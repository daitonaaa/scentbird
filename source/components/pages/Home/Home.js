import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';
import React, { Component } from 'react';

import './Home.scss';

import Controls from 'components/Controls';
import ItemsList from 'components/ItemsList';
import CreateForm from 'components/CreateForm';

import { getItemsList } from 'actions/items';


class Home extends Component {

  static propTypes = {
    count: PropTypes.number.isRequired,
    errorText: PropTypes.string.isRequired,

    getItemsList: PropTypes.func.isRequired,
  };

  static defaultProps = {
    count: 0,
    errorText: '',

    getItemsList: () => {}
  };

  componentDidMount() {
    this.props.getItemsList();
  }

  renderError() {
    const { errorText } = this.props;

    if (typeof errorText === 'string' && errorText.length) {
      return (
        <div className="items-list__error">
          {errorText}
        </div>
      );
    }
  }

  renderHeader() {
    const { count } = this.props;

    return (
      <div className="items-list__header">
        <div className="items-list__header-title">
          Пример мини приложения на React.JS
        </div>
        <div className="items-list__header-hint">
          <span>
            Код можно посмотреть тут -
          </span>
          <a href="https://github.com/Cast0001/code-example" target="_blank">
            Cast0001
          </a>
        </div>
        <div className="items-list__header-count">
          {`Выделено дочерних элементов: ${count}`}
        </div>
      </div>
    );
  }

  render() {
    return (
      <div className="home">
        <Helmet><title>Пример кода</title></Helmet>
        {this.renderError()}
        {this.renderHeader()}
        <Controls />
        <ItemsList />
        <CreateForm />
      </div>
    );
  }
}


const mapStateToProps = state => ({
  ...state.items,
});

const mapDispatchToProps = {
  getItemsList,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
