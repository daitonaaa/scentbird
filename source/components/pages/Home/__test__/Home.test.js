import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Home } from 'components/pages/Home/Home';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


describe('> > > COMPONENT - Home', () => {
  it('#create snapshot', () => {
    const output = shallow(
      <Home
        errorText=""
        request={false}
        getItemsList={()=>{}}
      />
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('#test render func => renderError', () => {
    const output = shallow(
      <Home
        errorText="something"
        request={false}
        getItemsList={()=>{}}
      />
    );

    expect(output.instance().renderError()).toBeDefined();
  });
});
