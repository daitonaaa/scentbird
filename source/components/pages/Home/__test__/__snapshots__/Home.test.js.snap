// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`> > > COMPONENT - Home #create snapshot 1`] = `
<div
  className="home"
>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Пример кода
    </title>
  </HelmetWrapper>
  <div
    className="home__header"
  >
    <div
      className="home__header-title"
    >
      Пример mini приложения на 
      <span>
        React.js
      </span>
    </div>
    <div
      className="home__header-hint"
    >
      <span>
        Код приложения можно посмотреть тут:
      </span>
      <a
        href="https://github.com/Cast0001/code-example"
        target="_blank"
      >
        GitHub
      </a>
    </div>
  </div>
  <div
    className="home__content"
  >
    <div
      className="home__content-left"
    >
      <Connect(Controls) />
    </div>
    <div
      className="home__content-right"
    >
      <Connect(ItemsList) />
    </div>
  </div>
  <Connect(CreateForm) />
</div>
`;
